### PR TITLE
[explorer] Adds Search Filter

### PR DESCRIPTION
**Pull request by Andrew47**
Fixes Issue https://github.com/MystenLabs/sui/issues/1871 through the use of a dropdown box. The UI has been optimised bearing in mind screens even as small as the Galaxy Fold.

image

image

### CODE REVIEW

**Comment by 666lcz** [comment]
Quick comment: is there a regression in the padding for search by ID or are you in a really old branch?

CleanShot 2022-05-13 at 11 47 41

**Comment by stella3d** [comment]
Added a small PR that changes the placeholder text for address and transaction categories, as shown here.

Also adds a TS type for defining the correct strings for categories

image

image

**Review by stella3d — APPROVED** [review]
tested this locally, works well, code looks good.  great job!

**Comment by 666lcz** [comment]
Please make sure fixing the padding issue before merging(see comment above)

File: explorer/client/src/components/search/Search.tsx [hunk 1]
```diff
@@ -8,8 +8,22 @@ import { navigateWithUnknown } from '../../utils/searchUtil';
 
 import styles from './Search.module.css';
 
+type SearchCategory = 'all' | 'transactions' | 'addresses' | 'objects';
+function getPlaceholderText(category: SearchCategory) {
+    switch (category) {
+        case 'addresses':
+            return 'Search by address';
+        case 'transactions':
+            return 'Search by digest';
```
Comment (666lcz) [inline explorer/client/src/components/search/Search.tsx:17]:
```suggestion
            return 'Search by transaction id';
```
Even though "digest" is technically more accurate, we use `TxnId` everywhere else, so let's be consistent here
CleanShot 2022-05-13 at 12 18 15

**Pull request merged**